README


This archive contains the distribution MediaWikiUtils,
version 0.141011:

  A tools provide few useful MediaWiki operation

This software is copyright (c) 2014 by Natal Ngétal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.009.



Changes
Revision history for MediaWikiUtils.

0.141011  2014-04-12 00:08:17CEST+0200 Europe/Paris

    * Add basic load test.
    * Forget commit.

0.141010  2014-04-11 23:45:49CEST+0200 Europe/Paris

    * Initial version developed.



Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.009.
use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "A tools provide few useful MediaWiki operation",
  "dist_author" => [
    "Natal Ng\x{e9}tal"
  ],
  "dist_name" => "MediaWikiUtils",
  "dist_version" => "0.141011",
  "license" => "perl",
  "module_name" => "MediaWikiUtils",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "LWP::UserAgent" => 0,
    "MediaWiki::API" => 0,
    "Moo" => 0,
    "MooX::Cmd" => 0,
    "MooX::Options" => 0,
    "Text::Unaccent::PurePerl" => 0,
    "Types::Standard" => 0,
    "pQuery" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [
    "bin/mwu"
  ],
  "test_requires" => {
    "Test::More" => 0
  }
);


my %fallback_build_requires = (
  "Module::Build" => "0.3601",
  "Test::More" => 0
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/MediaWikiUtils.pm
#
# This file is part of MediaWikiUtils
#
# This software is copyright (c) 2014 by Natal Ngétal.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package MediaWikiUtils;
{
  $MediaWikiUtils::VERSION = '0.141011';
}

use strict;
use warnings;

use Moo;
use MooX::Cmd;
use MooX::Options;

#ABSTRACT: A tools provide few useful MediaWiki operation

sub execute {
    shift->options_usage();
}


1;

__END__

=pod

=encoding UTF-8

=head1 NAME

MediaWikiUtils - A tools provide few useful MediaWiki operation

=head1 VERSION

version 0.141011

=head1 SYNOPSIS

    mwu

=head1 DESCRIPTION

Provides few useful command for mediawiki, like a mediawiki converter to
dokuwiki.

=head1 CONTRIBUTING

This module is developed on Github at:

L<http://github.com/hobbestigrou/MediaWikiUtils>

Feel free to fork the repo and submit pull requests

=head1 BUGS

Please report any bugs or feature requests in github.

=head1 SUPPORT

You can find documentation for this module with the perldoc command.

    perldoc MediaWikiUtils

=head1 SEE ALSO

L<MediaWiki::API>

=head1 AUTHOR

Natal Ngétal

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Natal Ngétal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/MediaWikiUtils/Common.pm
#
# This file is part of MediaWikiUtils
#
# This software is copyright (c) 2014 by Natal Ngétal.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package MediaWikiUtils::Common;
{
  $MediaWikiUtils::Common::VERSION = '0.141011';
}

use strict;
use warnings;

use Moo;
use MooX::Options;

use Types::Standard qw( Object );

use MediaWiki::API;
use LWP::UserAgent;

option 'username' => (
    is       => 'ro',
    format   => 's',
    required => 1,
    doc      => 'The username to login on the mediawiki'
);

option 'password' => (
    is       => 'ro',
    format   => 's',
    required => 1,
    doc      => 'The password to login on the mediawiki'
);

option 'url' => (
    is       => 'ro',
    format   => 's',
    required => 1,
    doc      => 'The url of the mediawiki'
);

has '_mediawiki' => (
    is       => 'lazy',
    isa      => Object,
    init_arg => undef
);

has '_user_agent' => (
    is       => 'lazy',
    isa      => Object,
    init_arg => undef
);

sub _build__mediawiki {
    my ( $self ) = @_;

    my $mediawiki = MediaWiki::API->new({
        api_url => $self->url
    });
    $mediawiki->login({
        lgname     => $self->username,
        lgpassword => $self->password
    }) || croak $mediawiki->{error}->{code} . ':' . $mediawiki->{error}->{details};

    return $mediawiki;
}

sub _build__user_agent {
    return LWP::UserAgent->new();
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

MediaWikiUtils::Common

=head1 VERSION

version 0.141011

=head1 AUTHOR

Natal Ngétal

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Natal Ngétal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/MediaWikiUtils/Convert.pm
#
# This file is part of MediaWikiUtils
#
# This software is copyright (c) 2014 by Natal Ngétal.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package MediaWikiUtils::Convert;
{
  $MediaWikiUtils::Convert::VERSION = '0.141011';
}

use strict;
use warnings;

use Moo;
use MooX::Cmd;
use MooX::Options;

use Carp;

use pQuery;
use Text::Unaccent::PurePerl qw(unac_string);

#ABSTRACT: A tools provide few method to convert MediaWiki to another wiki engine

option 'directory' => (
    is       => 'ro',
    format   => 's',
    default  => sub { '.' },
    doc      => 'The password to login on the mediawiki'
);

sub mediawiki2dokuwiki {
    my ( $self ) = @_;

    my $all_articles = $self->_mediawiki->list({
        action => 'query',
        list   => 'allpages'
    });

    foreach my $page (@{$all_articles}) {
        my $article = $self->_mediawiki->get_page({
            title => $page->{title}
        });

        my $title    = $page->{title};
        my $response = $self->_user_agent->post(
            'http://johbuc6.coconia.net/mediawiki2dokuwiki.php',
            [mediawiki => $article->{'*'}]
        );
        my $file = $self->_generate_file_name($title);

        if ( ! $response->is_success ) {
            carp "Request for $title failed: " . $response->status_line;
            next;
        }

        print 'Export the page ' . $page->{title} . " in the $file", "\n";
        pQuery($response->content)
            ->find('textarea[name=dokuwiki]')
            ->each(sub {
                my $count = shift;
                $self->_write_file($file, pQuery($_)->html());
        });
    }

    return;
}

sub _generate_file_name {
    my ( $self, $title ) = @_;

    $title =~  s/\s/_/g;
    $title =~  s/'/_/g;

    return unac_string($title) . '.txt';
}

sub _write_file {
    my ( $self, $file, $content ) = @_;

    my $path = $self->directory . '/' . $file;
    my $fh   = IO::File->new($path, "w");

    if ( defined($fh) ) {
        print $fh $content;

        undef $fh;
    }

    return;
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

MediaWikiUtils::Convert - A tools provide few method to convert MediaWiki to another wiki engine

=head1 VERSION

version 0.141011

=head1 AUTHOR

Natal Ngétal

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Natal Ngétal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/MediaWikiUtils/Cmd/ImportExport.pm
#
# This file is part of MediaWikiUtils
#
# This software is copyright (c) 2014 by Natal Ngétal.
#
# This is free software; you can redistribute it and/or modify it under
# the same terms as the Perl 5 programming language system itself.
#
package MediaWikiUtils::Cmd::ImportExport;
{
  $MediaWikiUtils::Cmd::ImportExport::VERSION = '0.141011';
}

use strict;
use warnings;

use Moo;
extends 'MediaWikiUtils::Common';

use MooX::Cmd;
use MooX::Options;

#ABSTRACT: A tools to provides command to import and export mediawiki data

option 'file' => (
    is       => 'ro',
    formart  => 's',
    default  => sub { 'wiki_dump.xml' },
    doc      => 'The file to use on the import (default wiki_dump.xml)'
);

sub execute {
    my ( $self ) = @_;

    $self->_mediawiki->api({
        action => 'import',
        xml    => [$self->file]
    });

    return;
}


1;

__END__

=pod

=encoding UTF-8

=head1 NAME

MediaWikiUtils::Cmd::ImportExport - A tools to provides command to import and export mediawiki data

=head1 VERSION

version 0.141011

=head1 SYNOPSIS

    mwu importexport --username mediawiki_user --password my_password --url http://mediawiki.com/api.php

=head1 DESCRIPTION

To export and import mediawiki to another mediawiki, at the moment the only
command supported is import.

=head1 AUTHOR

Natal Ngétal

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2014 by Natal Ngétal.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.009.
use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "A tools provide few useful MediaWiki operation",
  "AUTHOR" => "Natal Ng\x{e9}tal",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "MediaWikiUtils",
  "EXE_FILES" => [
    "bin/mwu"
  ],
  "LICENSE" => "perl",
  "NAME" => "MediaWikiUtils",
  "PREREQ_PM" => {
    "Carp" => 0,
    "LWP::UserAgent" => 0,
    "MediaWiki::API" => 0,
    "Moo" => 0,
    "MooX::Cmd" => 0,
    "MooX::Options" => 0,
    "Text::Unaccent::PurePerl" => 0,
    "Types::Standard" => 0,
    "pQuery" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Test::More" => 0
  },
  "VERSION" => "0.141011",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "LWP::UserAgent" => 0,
  "MediaWiki::API" => 0,
  "Module::Build" => "0.3601",
  "Moo" => 0,
  "MooX::Cmd" => 0,
  "MooX::Options" => 0,
  "Test::More" => 0,
  "Text::Unaccent::PurePerl" => 0,
  "Types::Standard" => 0,
  "pQuery" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



